cms/djangoapps/contentstore/views/tests/test_clipboard_paste.py

```diff
@@ -141,6 +143,69 @@ def test_copy_and_paste_component(self, block_args):
+        Test copying a unit (vertical) with tags from one course into another
+        """
+        course_key, client = self._setup_course()
+        dest_course = CourseFactory.create(display_name='Destination Course')
+        with self.store.bulk_operations(dest_course.id):
+            dest_chapter = BlockFactory.create(parent=dest_course, category='chapter', display_name='Section')
+            dest_sequential = BlockFactory.create(parent=dest_chapter, category='sequential', display_name='Subsection')
+
+        unit_key = course_key.make_usage_key("vertical", "vertical_test")
+        # Add tags to the unit
+        taxonomy_all_org = tagging_api.create_taxonomy("test_taxonomy", "Test Taxonomy")
+
+        tagging_api.set_taxonomy_orgs(taxonomy_all_org, all_orgs=True)
+        Tag.objects.create(taxonomy=taxonomy_all_org, value="tag_1")
+        Tag.objects.create(taxonomy=taxonomy_all_org, value="tag_2")
+        tagging_api.tag_object(
+            object_id=str(unit_key),
+            taxonomy=taxonomy_all_org,
+            tags=["tag_1", "tag_2"],
+        )
+
+        taxonomy_all_org_removed = tagging_api.create_taxonomy("test_taxonomy_removed", "Test Taxonomy Removed")
+        tagging_api.set_taxonomy_orgs(taxonomy_all_org_removed, all_orgs=True)
+        Tag.objects.create(taxonomy=taxonomy_all_org_removed, value="tag_1")
+        Tag.objects.create(taxonomy=taxonomy_all_org_removed, value="tag_2")
+        tagging_api.tag_object(
+            object_id=str(unit_key),
+            taxonomy=taxonomy_all_org_removed,
+            tags=["tag_1", "tag_2"],
+        )
+        tagging_api.get_object_tags(str(unit_key))
+
+        taxonomy_no_org = tagging_api.create_taxonomy("test_taxonomy_no_org", "Test Taxonomy No Org")
+        Tag.objects.create(taxonomy=taxonomy_no_org, value="tag_1")
+        Tag.objects.create(taxonomy=taxonomy_no_org, value="tag_2")
+        tagging_api.tag_object(
+            object_id=str(unit_key),
+            taxonomy=taxonomy_no_org,
+            tags=["tag_1", "tag_2"],
+        )
+
+        # Copy the unit
+        copy_response = client.post(CLIPBOARD_ENDPOINT, {"usage_key": str(unit_key)}, format="json")
+        assert copy_response.status_code == 200
+
+        taxonomy_all_org_removed.delete()
+
+        # Paste the unit
+        paste_response = client.post(XBLOCK_ENDPOINT, {
+            "parent_locator": str(dest_sequential.location),
+            "staged_content": "clipboard",
+        }, format="json")
+        assert paste_response.status_code == 200
+        dest_unit_key = UsageKey.from_string(paste_response.json()["locator"])
+
+        # Only tags from the taxonomy that is associated with the dest org should be copied
+        tags = list(tagging_api.get_object_tags(str(dest_unit_key)))
+        assert len(tags) == 2
+        assert str(tags[0]) == f'<ObjectTag> {dest_unit_key}: test_taxonomy=tag_1'
+        assert str(tags[1]) == f'<ObjectTag> {dest_unit_key}: test_taxonomy=tag_2'
```

Comment:
^ This test is great, and even if we end up changing how we handle the tags when copying and pasting, this test shouldn't need to be modified.